rtlib/shape_alph.hh

```diff
@@ -38,28 +47,34 @@ struct ShapeAlph {
   void operator()(T &t, char x, Size l) const {
     switch (x) {
       case '[' :
-        t |= T(1) << (l-1);
+        t |= T(1) << (l-(char_width-1));
         // set_zero(t, l);
         // set_one(t, l-1);
         break;
       case ']' :
-        t |= T(2) << (l-1);
+        t |= T(2) << (l-(char_width-1));
         // set_one(t, l);
         // set_zero(t, l-1);
         break;
       case '_' :
         // set_one(t, l);
         // set_one(t, l-1);
-        t |= T(3) << (l-1);
+        t |= T(3) << (l-(char_width-1));
+        break;
+      case 'G' :
+        // set_one(t, l);
+        // set_one(t, l-1);
+        t |= T(4) << (l-(char_width-1));
         break;
       default: assert(false);
     }
   }
   char to_char(T &t, Size i) const {
-    switch (t >> i & T(3)) {
+    switch (t >> i & T(pow(2, static_cast<int>(this->char_width))-1)) {
```

Comment:
Wouldn't it make sense to store the result of the `pow` call in a member variable since its basically a constant? This way the value gets recalculated every time this method is called, which probably won't affect execution time much but is kind of unnecessary nonetheless.

Reply:
true. I hope that `gcc`'s optimization routines would apply here, but I am not sure!

Reply:
I'm sure `gcc` can optimize something like this, but I have no idea if that's going to happen here. It should be able to detect this though.

Reply:
done. Better save than sorry ;-)